12.1-async-closure-to-access-a-child/Readme.md

```diff
@@ -0,0 +1,82 @@
+# 12.1 Async Closure
+
+A common use case for _useRef_ is to access a child imperatively.
+
+# Steps
+
+- We will take as starting point sample _00 boilerplate_. Copy the content of the
+  project to a fresh folder an execute _npm install_.
+
+```bash
+npm install
+```
+
+- Let's create an edge case, we will use _useRef_ to get the element clientWidth.
+  First, we need render a div element and create the ref with _useRef_.
+
+_./src/demo.js_
+
+```jsx
+import React from "react";
+
+export const MyComponent = () => {
+  const containerElementRef = React.useRef(null);
+
+  return <div className="container" ref={containerElementRef} />;
+};
+```
+
+\_./styles.css
+
+```diff
+.App {
+  font-family: sans-serif;
+  text-align: center;
+}
+
++.container {
++  border: 1px solid steelblue;
++  margin: 15px;
++  padding: 50px;
++}
+```
+
+- For this example, we show the container clientWidth. For render this element attribute,
+  we will create a handle button click to calculate it:
+
```

Comment:
In this example we will display the given container width, this action will be triggered when the user clicks on a button